streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -209,7 +207,9 @@ public void testStateOneThreadDeadButRebalanceFinish() throws InterruptedExcepti
 
         globalStreams.close();
 
-        Assert.assertEquals(6, stateListener.numChanges);
+        TestUtils.waitForCondition(
+            () -> stateListener.numChanges == 6,
+            "Streams never closed.");
```

Comment:
This is the actual fix: as there are two state transitions, we cannot simply check for six transitions, because if we check too early, and only five transitions finished (or maybe none), `stateListener.numChanges` might still be at 4 or 5.

Reply:
Thanks for the catch! It was my bad for not capturing it last time I modified this test case.